src/main/java/com/google/gwt/site/webapp/client/GWTProjectEntryPoint.java

```diff
@@ -363,75 +424,57 @@ private void onPageLoaded(boolean menuReplaced) {
    */
   private void scrollToHash() {
     String hash = Window.Location.getHash();
-    GQuery anchor = hash.length() > 1 ? $(hash + ", [name='" + hash.substring(1) + "']") : $();
-    if (anchor.isEmpty()) {
+    Element anchor = hash.length() > 1 ? document.querySelector(
+        hash + ", [name='" + hash.substring(1) + "']") : null;
+    if (anchor == null) {
       Window.scrollTo(0, 0);
     } else {
       anchor.scrollIntoView();
     }
   }
 
   private void maybeStyleHomepage() {
-    if ($("#content").hasClass("home")) {
+    if (document.querySelector("#content").classList.contains("home")) {
+      document.documentElement.style.setProperty("scroll-behavior", "smooth");
       styleHomepage();
-      resizeHandler = Window.addResizeHandler(new ResizeHandler() {
-        @Override
-        public void onResize(ResizeEvent event) {
-          styleHomepage();
-        }
-      });
+      resizeHandler = Window.addResizeHandler(event -> styleHomepage());
 
       // Pager
-      $(".next, .pager a").click(new Function() {
-        @Override
-        public boolean f(final Event event) {
-          event.preventDefault();
-          $("html, body")
-              .each(new Function() {
-                @Override
-                public void f(Element e) {
-                  new ScrollTopAnimation(e, getElementOffset($($(event).attr("href")))).run(600);
-                }
-              });
-          return true;
-        }
-      });
-
-      $(".pager a").click(new Function() {
-        @Override
-        public boolean f(Event e) {
-          e.preventDefault();
-          $(".pager a").removeClass("active");
-          $(this).addClass("active");
-
-          return true;
-        }
-      });
-
-      $(window).scroll(new Function() {
-        @Override
-        public void f() {
-          $(".pager a").removeClass("active");
-          if ($(window).scrollTop() + 100 > getElementOffset($("#letsbegin"))) {
-            $(".pager a:nth-child(3)").addClass("active");
-          } else if ($(window).scrollTop() + 100 > getElementOffset($("#gwt"))) {
-            $(".pager a:nth-child(2)").addClass("active");
+      forEach(".next, .pager a", (element) -> element.addEventListener("click", event -> {
+        event.preventDefault();
+        document.documentElement.scrollTop =
+           getElementOffset(document.querySelector(element.getAttribute("href")));
+      }));
+
+      forEach(".pager a", (element) -> element.addEventListener("click", event -> {
+        event.preventDefault();
```

Comment:
note that the above forEach already called preventDefault() on all click events on `.pager a` items